tests/integration/test_machine.py

```diff
@@ -36,3 +37,39 @@ async def test_status():
                          machine.status_message.lower() == 'running' and
                          machine.agent_status == 'started')),
             timeout=480)
+
+
+@base.bootstrapped
+async def test_scp(event_loop):
+    # ensure that asyncio.subprocess will work;
+    try:
+        asyncio.get_child_watcher().attach_loop(event_loop)
+    except RuntimeError:
+        pytest.skip('test_scp will always fail outside of MainThread')
+    async with base.CleanModel() as model:
+        await model.add_machine()
+        await asyncio.wait_for(
+            model.block_until(lambda: model.machines),
+            timeout=240)
+        machine = model.machines['0']
+        await asyncio.wait_for(
+            model.block_until(lambda: (machine.status == 'running' and
+                                       machine.agent_status == 'started')),
+            timeout=480)
+
+        with NamedTemporaryFile() as f:
+            f.write(b'testcontents')
+            f.flush()
+            await machine.scp_to(f.name, 'testfile', scp_opts='-p')
+
+        with NamedTemporaryFile() as f:
+            await machine.scp_from('testfile', f.name, scp_opts='-p')
+            assert f.read() == b'testcontents'
+
+
+async def test_machine_ssh():
+    async with base.CleanModel() as model:
+        machine: Machine = await model.add_machine()
+        out = await machine.ssh("echo hello world!")
```

Comment:
Same redundancy here, see the `test_ssh` test in [integration/test_unit.py](https://github.com/juju/python-libjuju/blob/16149dbef02160a7f4170e1630bfa069ba275479/tests/integration/test_unit.py#L193), `unit.ssh` is just a wrapper around the `machine.ssh`.

Reply:
Added `wait_for_active` parameter here - this should no longer be redundant :)